### DISCUSSION TITLE
3W as machine learning classification problem

### DISCUSSION THREAD

**Discussion by Shaima3j** [petrobras/3W#88] [General]
Good day,

I have tried to use 3W data, in machine learning classifications project.
I focused on faults 1, 2, 5 real time data.
When testing the model on separate dataset (one from each folder was separated from training and kept for testing).
The model could only detect class 0. (Even after applying techniques that solve imbalance classes issue)
Is there any logical reason behind this? Are there data overlapping?
is data in certain date totally different than other date?
I am trying to figure out the reason with no luck.

Because there is no mathematical relationship between columns and labels, I am unable to detect why most data considered by machine learning as normal.

**Comment by Shaima3j** [comment]
Is there any common pattern or any indication that I can rely on?

**Comment by afraniomelo** [comment]
Hello,

I suggest taking a closer look at the data in order to better visualize the underlying patterns. For this purpose, the Exploratory Data Analysis (EDA) notebooks accessible on [Kaggle](https://www.kaggle.com/datasets/afrniomelo/3w-dataset) could provide valuable insights. Additionally, you might consider applying feature engineering techniques to accentuate the patterns in data fed to classification models. To delve deeper into this subject, a comprehensive list of references can be found [here](https://github.com/petrobras/3W/blob/main/LIST_OF_CITATIONS.md).
